TaffoInitializer/IndirectCallPatcher.cpp

```diff
@@ -0,0 +1,174 @@
+#include "IndirectCallPatcher.h"
+
+#include "Metadata.h"
+#include "TaffoInitializerPass.h"
+#include "TypeUtils.h"
+#include "llvm/ADT/Statistic.h"
+#include "llvm/IR/Constants.h"
+#include "llvm/IR/Function.h"
+#include "llvm/IR/InstIterator.h"
+#include "llvm/IR/Instructions.h"
+#include "llvm/IR/Module.h"
+#include "llvm/Support/CommandLine.h"
+#include "llvm/Support/Debug.h"
+#include <llvm/IR/IRBuilder.h>
+#include <llvm/Transforms/Utils/Cloning.h>
+#include <llvm/Transforms/Utils/ValueMapper.h>
+
+using namespace taffo;
+using namespace llvm;
+
+/// Check recursively whether an unsupported function is called.
+bool containsUnsupportedFunctions(llvm::Function *function, int exploreDepth) {
+  std::vector<std::string> prefixBlocklist{"__kmpc_omp_task"};
+
+  for (auto instructionIt = inst_begin(function);
+       instructionIt != inst_end(function); instructionIt++) {
+    if (auto curCallInstruction = dyn_cast<CallInst>(&(*instructionIt))) {
+      llvm::Function *curCallFunction = curCallInstruction->getCalledFunction();
+      auto functionName = curCallFunction->getName();
+
+      if (any_of(prefixBlocklist, [&](const std::string &prefix) {
+            return functionName.startswith(prefix);
+          })) {
+        return true;
+      } else if (exploreDepth > 0) {
+        if (containsUnsupportedFunctions(curCallFunction, exploreDepth - 1)) {
```

Comment:
To be honest, instead of using a depth counter to prevent infinite recursion, I would prefer to use a set of traversed functions.

Reply:
Used an unordered_set of functions and going to add dedicated tests to verify the behaviour.